students/tri_nguyen/lesson07/html_render.py

```diff
@@ -0,0 +1,159 @@
+# ------------------------------------------- #
+# Student:   Tri Nguyen
+# Instructor: Natasha Aleksandrova
+# Date:  23-Mar-2018
+# ------------------------------------------- #
+
+# Create an Element class
+
+
+class Element():
+    ''' Element class for rendering an html element '''
+
+    # Class attributes
+    tag_name = ''
+    indent = '    '  # 4 spaces
+
+    def __init__(self, content=None, **kwargs):
+        ''' initialize the Element object/instance '''
+        if content is None:
+            self.content = []
+        else:
+            self.content = [content]
+        self.kwargs = kwargs
+
+    def append(self, content):
+        ''' store content to self.content list '''
+        self.content.append(content)
+
+    def render(self, file_out, cur_ind=''):
+        ''' render the tag and the strings in the content '''
+        file_out.write('{}<{}{}>\n'.format(cur_ind, self.tag_name, self.set_attr()))
+
+        '''for attr in self.kwargs:
+            file_out.write(' {}="{}"'.format(attr, self.kwargs[attr]))
+        file_out.write('>\n')'''
+
+        for item in self.content:
+            if hasattr(item, 'render'):
+                item.render(file_out, cur_ind + self.indent)
+            else:
+                file_out.write('{}{}{}\n'.format(cur_ind, self.indent, item))
+
+        file_out.write('{}</{}>\n'.format(cur_ind, self.tag_name))
+
+    def set_attr(self):
+        if self.kwargs:
+            for k, v in self.kwargs.items():
+                attr, val = k, v
+            return ' {}="{}"'.format(attr, val)
```

Comment:
On line 212 of `run_html_render.py`, both an `id` and `style` attribute get passed into the `Ul` constructor, but I only see the `style` attribute in your rendered html output. Can you see where you're losing the `id` attribute in the the `set_attr()` method above and correct?